Add an image container component that displays all saved images

jbewing opened the pull request:
This PR adds a first cut at an image container component. I learned Typescript + refreshed myself on React and this is what resulted. All pointers + feedback is appreciated as I don't usually jump into the front of the stack that often 😆 

**Preview**: 
<img width="1048" alt="Screen Shot 2020-10-28 at 11 42 49 AM" src="https://user-images.githubusercontent.com/13725922/97460045-c7601280-1912-11eb-93fb-9f65aa19d2cb.png">

_Notes_: 
- I'm sure that everything could be flex-boxed up and the margins could be automatically adjusted so that everything is centered, but it would take a better CSS mind than mine. I just left everything as `inline-block` + left justified.

- There was a bug turned out to be because of CORS, so I added a CORS module to the API which seems to have fixed it. Please tell me if this is not the desired approach + what could be done better. The end goal is just to be able to read the API responses from JS instead of having things fail in weird ways behind the scenes but I usually don't need to worry about this stuff as a BE dev 😄 

- The tooling is amazing! Thanks for setting that up @DarkAce65 

### Testing
I tested this component with the flask server running locally (`poetry run task run`) + the FE running (`yarn start`). I uploaded a ton of duplicate copies of a golden retriever to the DB to test how this component acted with 1 image vs. three rows of images. I tested that the design is  responsive to screen width + height as well.

Issue: https://github.com/brodigan-e/capstone-POV/issues/11
cc @DarkAce65

jbewing committed:
Add first cut at an image container

DarkAce65 commented on frontend/src/containers/ImageContainer.tsx:
```diff
@@ -0,0 +1,41 @@
+  }
+
+  componentDidMount() {
+    fetch(IMAGE_API, { mode: 'cors' })
+      .then((res) => res.json())
```
tiny nit: I would move this into a function that returns a Promise (maybe in `src/data/something.ts`) so that components don't need to care about the fetch internals and only handle the responses

DarkAce65 commented on frontend/src/components/ImageComponent.tsx:
```diff
@@ -0,0 +1,39 @@
+
+const ImageComponent: FunctionComponent<Props> = ({ image }) => {
+  return (
+    <TilingFigure>
```
nit: I'd probably put the `TilingFigure` styling in the `ImageContainer` because the `ImageComponent` on its own shouldn't be concerned about layout outside itself

DarkAce65 reviewed (approved):
Nice! I just left a few tiny comments but this looks really good :smile:

DarkAce65 commented on frontend/src/containers/ImageContainer.tsx:
```diff
@@ -0,0 +1,41 @@
+  }
+
+  componentDidMount() {
+    fetch(IMAGE_API, { mode: 'cors' })
```
one more tiny thing: `mode: 'cors'` is the default setting so you can probably just omit this second argument

jbewing committed:
PR feedback

- Split out image fetching action to own file
- Add antd and redo some styling based on it
-remove default 'cors' mode

jbewing commented:
Lemme quickly resolve these conflicts, then I'd appreciate a re-review 😄  @DarkAce65

jbewing committed:
Merge branch 'master' into typescript-adventures

jbewing commented on frontend/src/components/ImageComponent.tsx:
```diff
@@ -0,0 +1,33 @@
+  border: 1px solid #ddd;
+  border-radius: 0.15em;
+  padding: 0.15em;
+`;
```
This one _could_ be converted to the `antd` equivalent. Wanted to know what the convention is for this custom-ish stuff. Similarly for the styled caption below, I really have no idea the best way to tidy this up i.e. as is versus

```javascript
import { Image } from 'antd';
...
<figure>
    <Image src={API_SERVER + image.href} width="100%" height="100%" alt={image.title} />
    <CenteredCaption>{image.title}</CenteredCaption>
</figure>
```

DarkAce65 replied:
Yeah this could probably be a `Card` from Ant ([docs](https://ant.design/components/card/#components-card-demo-flexible-content)). There's no real convention for this kind of stuff but images with captions are often just stuffed into cards. The `figure` and `figcaption` elements are more for the article context (bit like LaTeX in that regard) but there's no hard and fast rule for that

jbewing commented on frontend/src/data/ImagesClient.ts:
```diff
@@ -0,0 +1,9 @@
+  return fetch(IMAGE_API).then((res) => res.json());
+}
+
+export { getImages };
```
As a BE dev, this was the only reasonable solution I could come up with: writing a client.

DarkAce65 replied:
This is perfect :smile:

jbewing commented on frontend/src/containers/ImageContainer.tsx:
```diff
@@ -0,0 +1,45 @@
+          ))}
+        </Row>
+      </>
+    );
```
Is "clean" as far as React is concerned? I'm not really sure how to balance for-loop complexity stuff in `render()`. Should it be pulled out or is it fine to keep it as is if it's _simple-ish_

DarkAce65 replied:
Yeah, that looks good to me (though the fragment wrapping the `Row` isn't necessary anymore). My personal opinion for the for-loops in the render function is that it's fine as long as it's just a `map` - anything that's more complicated than a simple mapping probably deserves its own function.

DarkAce65 commented on frontend/src/data/ImagesClient.ts:
```diff
@@ -0,0 +1,9 @@
+
+import { IMAGE_API } from './ApiConstants';
+
+async function getImages(): Promise<Array<Image>> {
```
tiny nit: `async` is not needed here since there's no `await` in the function

jbewing committed:
PR feedback

- Get rid of styled components in favor antd
- Remove async
